docs/source/common_issues.rst

```diff
@@ -821,3 +821,32 @@ This is best understood via an example:
 To get this code to type check, you could assign ``y = x`` after ``x`` has been
 narrowed, and use ``y`` in the inner function, or add an assert in the inner
 function.
+
+
+Upgrading to PEP584 syntax
+---------------------------
+
+In pre-PEP584 syntax type annotations for builtin types such as ``dict``, ``list``, ``tuple``, ``type`` were taken from the ``typing`` module as ``typing.Dict``, ``typing.List``, ``typing.Tuple``, ``typing.Type`` etc.
+
+When upgrading a codebase you may hit a case such as the following:
+
+.. code-block:: python
+
+    from typing import TypeVar
+
+    T = TypeVar('T', bound='Foo')
+
+    class Foo:
+	type: str
+
+	@classmethod
+	def foo(cls: type[T], value: str) -> None:  # will fail due to "not a valid type"
```

Comment:
- It's somewhat confusing for the method to be called `foo()` when the class is called `Foo`.
- Be slightly more specific about mypy's complaint here

```suggestion
	def bar(cls: type[T], value: str) -> None:  # Mypy complains that "Foo.type" is "not valid as a type"
```

Reply:
I moved the comment here to the pass statement to not make it above 80 chars and have the sliding bar appear on the bottom of the code block. I am not sure if this is preferred. Let me know.